### PR TITLE
micp

### PR DESCRIPTION
**Pull request by daavid00**
It depends on: OPM/opm-common#2743 OPM/opm-material#475 OPM/opm-models#671

### CODE REVIEW

File: ebos/eclproblem.hh [hunk 1]
```diff
@@ -1194,9 +1212,10 @@ public:
     {
         // use the generic code to prepare the output fields and to
         // write the desired VTK files.
-        ParentType::writeOutput(verbose);
 
         bool isSubStep = !EWOMS_GET_PARAM(TypeTag, bool, EnableWriteAllSolutions) && !this->simulator().episodeWillBeOver();
+        if (!isSubStep)
+            ParentType::writeOutput(verbose);
         if (enableEclOutput_)
```
Comment (totto82) [inline ebos/eclproblem.hh:1219]: Can you comment on this change?

File: opm/simulators/flow/BlackoilModelEbos.hpp [hunk 2]
```diff
@@ -451,13 +465,13 @@ namespace Opm {
 
                 Scalar saturationsNew[FluidSystem::numPhases] = { 0.0 };
                 Scalar oilSaturationNew = 1.0;
-                if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx)) {
+                if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx) && !has_micp_) {
                     saturationsNew[FluidSystem::waterPhaseIdx] = priVarsNew[Indices::waterSaturationIdx];
```
Comment (totto82) [inline opm/simulators/flow/BlackoilModelEbos.hpp:469]: Is this has_micp_ related to issues with running one-phase only water?

File: opm/simulators/wells/WellInterface_impl.hpp [hunk 3]
```diff
@@ -61,7 +61,6 @@ namespace Opm
                 }
             }
         }
-        well_control_log_.clear();
     }
```
Comment (totto82) [inline opm/simulators/wells/WellInterface_impl.hpp:64]: This seems unrealted

**Review by totto82 — COMMENTED** [review]
Thanks for the contribution. In general the code looks good. I have some comments that needs to be addressed before it can be merged.

File: opm/simulators/flow/BlackoilModelEbos.hpp (hunk 2 above)
Comment (daavid00) [reply opm/simulators/flow/BlackoilModelEbos.hpp:469]: Yes, I have made it more general to include all single-phase with water cases, see latest commit.

**Comment by totto82** [comment]
jenkins build this opm-common=2743 opm-material=475 opm-models=671 please

**Comment by totto82** [comment]
jenkins build this opm-common=2743 opm-models=671 please

**Comment by totto82** [comment]
jenkins build this opm-common=2743 opm-models=671 please

**Comment by atgeirr** [comment]
jenkins build this opm-common=2743 opm-models=671 please

**Comment by atgeirr** [comment]
jenkins build this opm-common=2743 opm-models=671 please

**Pull request merged**